packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -245,10 +246,72 @@ class SnackBar extends StatelessWidget {
   /// The animation driving the entrance and exit of the snack bar.
   final Animation<double> animation;
 
+  /// Called the first time that the snackbar is visible within a [Scaffold].
+  final VoidCallback onVisible;
+
+  // API for Scaffold.showSnackBar():
+
+  /// Creates an animation controller useful for driving a snack bar's entrance and exit animation.
+  static AnimationController createAnimationController({ @required TickerProvider vsync }) {
+    return AnimationController(
+      duration: _snackBarTransitionDuration,
+      debugLabel: 'SnackBar',
+      vsync: vsync,
+    );
+  }
+
+  /// Creates a copy of this snack bar but with the animation replaced with the given animation.
+  ///
+  /// If the original snack bar lacks a key, the newly created snack bar will
+  /// use the given fallback key.
+  SnackBar withAnimation(Animation<double> newAnimation, { Key fallbackKey }) {
+    return SnackBar(
+      key: key ?? fallbackKey,
+      content: content,
+      backgroundColor: backgroundColor,
+      elevation: elevation,
+      shape: shape,
+      behavior: behavior,
+      action: action,
+      duration: duration,
+      animation: newAnimation,
+      onVisible: onVisible,
+    );
+  }
+
+  @override
+  State createState() => _SnackBarState();
```

Comment:
missing a generic type annotation on State

Reply:
surprised the analyzer didn't catch it

Reply:
Done

Reply:
I wonder if this is because of the optionTypeArgs annotation:

https://github.com/flutter/flutter/blob/f0553ba58e6455aa63fafcdca16100b81ff5c3ce/packages/flutter/lib/src/widgets/framework.dart#L936